src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -67,113 +69,143 @@ public static ParseUDFsResult ParseUDFsScript(string script, CultureInfo loc = n
-                    var varIdent = TokEat(TokKind.Ident);
-                    ParseTrivia();
-                    if (TokEat(TokKind.Colon) == null)
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingColon);
-                        break;
-                    }
-
-                    ParseTrivia();
-                    var varType = TokEat(TokKind.Ident);
-                    ParseTrivia();
-                    if (args.ContainsKey(varIdent.ToString()))
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_DuplicateArgName);
-                        return new ParseUDFsResult(udfs, _errors);
-                    }
+            return new ParseUDFsResult(udfs, _errors);
+        }
 
-                    args.Add(varIdent.ToString(), new UDFArg(varIdent.As<IdentToken>(), varType.As<IdentToken>()));
-                    if (_curs.TokCur.Kind != TokKind.ParenClose && _curs.TokCur.Kind != TokKind.Comma)
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingComma);
-                        return new ParseUDFsResult(udfs, _errors);
-                    }
-                    else if (_curs.TokCur.Kind == TokKind.Comma)
-                    {
-                        TokEat(TokKind.Comma);
-                    }
-                }
+        private bool ParseUDF(List<UDF> udfs)
+        {
+            // <udf> ::= IDENT '(' <args> ')' ':' IDENT '=>' <function-body>
+            ParseTrivia();
+            var ident = TokEat(TokKind.Ident);
+            if (ident == null)
+            {
+                return false;
+            }
 
-                TokEat(TokKind.ParenClose);
+            var args = new Dictionary<string, UDFArg>();
+            if (TokEat(TokKind.ParenOpen) == null)
+            {
+                return false;
+            }
 
+            while (_curs.TokCur.Kind != TokKind.ParenClose)
+            {
+                ParseTrivia();
+                var varIdent = TokEat(TokKind.Ident);
                 ParseTrivia();
-
                 if (TokEat(TokKind.Colon) == null)
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingColon);
                     break;
                 }
 
                 ParseTrivia();
-
-                var returnType = TokEat(TokKind.Ident);
-                if (returnType == null)
+                var varType = TokEat(TokKind.Ident);
+                ParseTrivia();
+                if (args.ContainsKey(varIdent.ToString()))
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingIdentifier);
-                    break;
+                    return false;
                 }
 
-                ParseTrivia();
-
-                if (TokEat(TokKind.Equ) == null)
+                args.Add(varIdent.ToString(), new UDFArg(varIdent.As<IdentToken>(), varType.As<IdentToken>()));
```

Comment:
Consider keeping the tokens around, instead of stringifying them, so we could create errors on them later if needed.